src/extension/wps/wps-core/src/main/java/org/geoserver/wps/executor/WPSExecutionManager.java

```diff
@@ -391,6 +391,13 @@ private ExecuteResponseType execute() {
                 SubProgressListener executionListener = new SubProgressListener(listener,
                         inputPercentage, executionPercentage);
                 notifier.checkDismissed();
+                
+                //The system only moves from Queued to Running when the progress notifier fires (!!!).  This fires when inputs are read.
+                //   However, if there are no inputs, it never moves to running state!
+                //Since we are executing, we can move this from queued to running state. 
+                if (status.getPhase() == ProcessState.QUEUED) {
```

Comment:
Please keep comment professional (in general, no exclamations, no personal opinions and the like) and current to the actual state of things, like, if this one fixes the code not moving the process to running state, then don't mention it does not move there.